include/tc/core/cuda/cuda_execution_engine.h

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <dlpack/dlpack.h>
+
+#include "tc/core/cuda/cuda_tc_executor.h"
+#include "tc/core/execution_engine.h"
+#include "tc/core/mapping_options.h"
+
+namespace tc {
+
+/// The goal for this new shiny API is to provide a different pathway for being
+/// able to execute the kernels for multiple TC i.e. given the language which
+/// can have multiple TCs, people should be able to run things by just calling
+/// out the run function with the name of function and the inputs to run on.
+class CudaExecutionEngine : public ExecutionEngine {
+ public:
+  struct CudaExecutorInfo : public ExecutionEngine::ExecutorInfo {
+    CudaExecutorInfo(
+        std::string id,
+        std::vector<const DLTensor*> inputsInfo,
+        const MappingOptions& options,
+        lang::TreeRef tc,
+        size_t handle)
+        : ExecutionEngine::ExecutorInfo(
+              id,
+              inputsInfo,
+              options.toProtobufSerializedString(),
+              tc,
+              handle) {
+      exec =
+          std::unique_ptr<CudaTcExecutor>(new CudaTcExecutor(tc, inputsInfo));
+    }
+
+    void clear() {
+      static_cast<CudaTcExecutor&>(*exec).clearRTC();
+    }
+  };
+
+  CudaExecutionEngine() = default;
+
+  void addTC(const std::string& tc);
+
+  lang::TreeRef treeForFunction(const std::string& name) {
```

Comment:
can we name it `treeForDef`? I find `def` to directly related to def in the TC and it becomes clear what's going on here.

Reply:
yes, in a separate PR